aca_lts_eval.py

```diff
@@ -176,10 +177,45 @@ def select_ri_guide_stars(ra, dec, cone_stars):
     return G_RI_CAT_CACHE[id_key]
 
 
-def t_lose_star(mag, ref_mag=10.3, ref_tccd=-11.5, scale_4c=None):
+def guide_count(mags, tccd=-10.2):
+    """
+    Given mags from guide stars and a temperature, calculate a guide star
+    count using signal-to-noise scaled mag thresholds.
+    """
+    thresh1 = snr_mag_for_tccd(tccd, ref_mag=10.0)
+    thresh2 = snr_mag_for_tccd(tccd, ref_mag=10.2)
+    thresh3 = snr_mag_for_tccd(tccd, ref_mag=10.3)
+    counts = np.zeros_like(mags)
+    counts[mags <= thresh1] = 1.0
```

Comment:
Needs to have the `thresh0` lower limit included.  Below 6.0 mag the count is 0.0.  Or maybe 6.0 should be 5.9 based on my analysis that the observed mag err around 6 mag is 0.1 mag, and the guide star stops being useful at 5.8.

BTW, probably want to put the same into starcheck.

https://github.com/sot/starcheck/blob/fa0eaad500eef13f13c611d3e9bddd5504f9f7e8/starcheck/src/lib/Ska/Starcheck/Obsid.pm#L2548

Or else consider modifying starcheck to call this Python function here since the Perl code doesn't do temperature dependence.  (Would this get things out of sync with SAUSAGE?).

Reply:
"Needs to have the thresh0 lower limit included."

Sure, I can do that but I'm not really seeing a driver there.  They shouldn't be valid guide stars to be counted (and this routine is specific to guide stars).  Seems like we should just tweak the mini-sausage selection so they don't get picked for now (and I suppose that *does* allow selection up to 5.8 with err).

Reply:
"Or else consider modifying starcheck to call this Python function here since the Perl code doesn't do temperature dependence. (Would this get things out of sync with SAUSAGE?)."

Right, I thought we explicitly did't want to do temperature dependent counts for official load review and make that a mismatch and get out of sync with ORViewer.   Though with the reference temperature set at -10.2, I suppose we'd only get out of sync in a way that we really care about for observations that are hotter than the planning limit.  This probably deserves more thought.

In general, my thought with including the temperature dependent counts for aca_lts_eval is that it gives us something reasonable to steer the planning process a little (a gentle nudge toward a better bin if possible and acknowledgement that the field is borderline or ACA constrained at the OR list level), but it wouldn't create any mismatches during the load review process.

Reply:
Agreed on not letting starcheck get out of sync with SAUSAGE, but something to keep in mind because this count metric really should be a function of temperature and we are only OK right now because temps happen to be around -10.  For a planning limit of -5 then this implementation will be no good, but presumably it won't be in play by then anyway.

On the low limit, the driver is to make the metric and checking be robust against *anything* that gets thrown in.  It can't happen in real life, but imagine a field with all 5.85 mag stars.  This should get dinged by the ACA LTS eval as extremely difficult to schedule, while one with all 6.05 mag stars is A-OK.  The real SAUSAGE will pick 5.8 guide stars now, and there is no code that will prevent hand-edits of any magnitude.  So starcheck should also be dinging stars brighter than 5.9.

Reply:
"On the low limit, the driver is to make the metric and checking be robust against anything that gets thrown in."

I'm with you, I just have control of the *anything* from the minisausage now and you'll have control of it from proseco.  That was my point.  I've already got minisausage doing minor improvements over SAUSAGE with the idea that what the tool is really reporting is the constraining temperature of the best catalog we could easily pick manually at this attitude. 

So yes, for an OR first imagine a field with 5 5.85 stars.

If we update the thresh0 that will get a temperature of -16 out of the tool (just gets set to low limit).
If we update minisausage to not pick the stars brighter than 6, there will be no stars so it will get assigned a temperature of -16 as well.

Now let's imagine that field with 5 5.8 stars and 5 7.0 mags stars too.
If we just set thresh0, our answer for the field is -16C.
If we update minisausage with the 6.0 mag, the 5 too bright stars are discarded and the OK ones are picked, and our answer is +5C.
I think updating minisausage is more helpful to the overall process.  

Though of course adding a thresh0 and updating minisausage with a better bright limit for guide aren't mutually exclusive.

Reply:
Also, do you want to move this guide_count to chandra_aca.star_probs or some such (in which case I'll be fine thinking of it more generally) or do you want to keep this in our back pocket for starcheck when we decide to go that way?

Reply:
I was thinking of `guide_count` being a system-wide function that is not tightly bound to aca_lts_eval.  So it would be serving as the primary implementation of a concept that was agreed to by SSAWG.  If we chose to, it could be called by ORviewer, starcheck, starhopper, Jupyter notebooks, whatever.  In that paradigm of re-usable code, it makes sense to have a low limit of 5.9.  And it would also make sense to factor it out to `chandra_aca.star_probs` (since it is returning a probabilistic estimate of guide star count).

Agreed that you should update mini-sausage to not pick too-bright stars, that makes sense.

Reply:
And agreed that we already agreed to not use this right now for starcheck, but you *should* update the starcheck perl code to use a low limit so too-bright stars are not included in the guide count.

Reply:
Good idea.  They at least are already getting "Critical" warnings.